contrib/buildinfo/src/mill/contrib/buildinfo/BuildInfo.scala

```diff
@@ -1,58 +1,200 @@
 package mill.contrib.buildinfo
 
-import mill.T
-import mill.Agg
-import mill.api.{Logger, Loose, PathRef}
-import mill.define.Target
-import mill.scalalib.{Dep, ScalaModule, DepSyntax}
+import mill.{PathRef, T}
+import mill.scalalib.{JavaModule, ScalaModule}
 
-trait BuildInfo extends ScalaModule {
-
-  def buildInfoPackageName: Option[String] = None
+trait BuildInfo extends JavaModule {
+  /**
+   * The package name under which the BuildInfo data object will be stored.
+   */
+  def buildInfoPackageName: String
 
+  /**
+   * The name of the BuildInfo data object, defaults to "BuildInfo"
+   */
   def buildInfoObjectName: String = "BuildInfo"
 
-  def buildInfoMembers: T[Map[String, String]] = T {
-    Map.empty[String, String]
+  /**
+   * Enable to compile the BuildInfo values directly into the classfiles,
+   * rather than the default behavior of storing them as a JVM resource. Needed
+   * to use BuildInfo on Scala.js which does not support JVM resources
+   */
+  def buildInfoStaticCompiled: Boolean = false
+
+  /**
+   * A mapping of key-value pairs to pass from the Build script to the
+   * application code at runtime.
+   */
+  def buildInfoMembers: T[Seq[BuildInfo.Value]]
```

Comment:
We should provide a default of `Seq.empty` here. This better works with stacked traits, as it allows the use of `super`-call. We already handle the empty Seq.

Reply:
I don't quite understand this point. How does providing a default allow traits to stack better than otherwise?

Reply:
IIRC, if there is no default, you can't call `super.buildInfoMembers`, as super is abstract.

Reply:
done